framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1081,7 +1086,47 @@ protected void printOrStoreMessage(
             String message,
             Tree source,
             CompilationUnitTree root) {
+        StackTraceElement[] trace = Thread.currentThread().getStackTrace();
+        printOrStoreMessage(kind, message, source, root, trace);
+    }
+
+    /**
+     * printOrStoreMessage method with an added stack trace argument. The stack trace is printed
+     * when the dumpOnErrors option is enabled.
+     *
+     * @param kind the kind of message to print
+     * @param message the message text
+     * @param source the souce code position of the diagnostic message
+     * @param root the compilation unit
+     * @param trace the stack trace where the checker encountered an error
+     */
+    protected void printOrStoreMessage(
+            javax.tools.Diagnostic.Kind kind,
+            String message,
+            Tree source,
+            CompilationUnitTree root,
+            StackTraceElement[] trace) {
+        // Not calling printOrStoreMessage(kind, message, source, root) since the error message
```

Comment:
That comment is out-of-date - you would now create an infinite loop, so this question no longer arises.

Reply:
I have deleted the comment in my latest commit.